skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,322 @@
+            )
+            .addParameter(
+              ParameterSpec.builder(
+                  "eventSink",
+                  LambdaTypeName.get(
+                    receiver = eventInterfaceName,
+                    returnType = Unit::class.asTypeName(),
+                  ),
+                )
+                .defaultValue("{}")
+                .build()
+            )
+            .build()
+        )
+        .addProperty(PropertySpec.builder("message", String::class).initializer("message").build())
+        .addProperty(
+          PropertySpec.builder(
+              "eventSink",
+              LambdaTypeName.get(
+                receiver = eventInterfaceName,
+                returnType = Unit::class.asTypeName(),
+              ),
+            )
+            .initializer("eventSink")
+            .build()
+        )
+        .addSuperinterface(circuitUiState)
+        .build()
+
+    return FileSpec.builder(packageName, circuitScreenClass)
+      .addType(
+        TypeSpec.classBuilder(circuitScreenClass)
+          .addSuperinterface(screenInterface)
+          .addAnnotation(parcelize)
+          .addType(stateClass)
+          .addType(eventInterface)
+          .build()
+      )
+      .build()
+  }
+}
+
+class CircuitPresenter(
+  private val useAssistedInjection: Boolean = true,
+  private val noUi: Boolean = false,
+) : CircuitComponent {
+  override val fileSuffix = "Presenter"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /*
+     Generate Circuit Presenter class, eg:
+
+     class FeaturePresenter @AssistedInject constructor(
+       @Assisted
+       private val screen: {Feature}Screen,
+       @Assisted
+       private val navigator: Navigator,
+     ) : Presenter<ggScreen.State> {
+       @Composable
+       override fun present(): FeatureScreen.State {
+         val scope = rememberStableCoroutineScope()
+         return FeatureScreen.State() { event ->
+         }
+       }
+
+       @AssistedFactory
+       @CircuitInject(
+         FeatureScreen::class,
+         UserScope::class,
+       )
+       interface Factory {
+         fun create(screen: ggScreen, navigator: Navigator): FeaturePresenter {
+         }
+       }
+     }
+    */
+    val screenClassName = "${className}Screen"
+    val presenterClassName = "${className}Presenter"
+    val assistedInject = ClassName("dagger.assisted", "AssistedInject")
+    val assisted = ClassName("dagger.assisted", "Assisted")
+    val assistedFactory = ClassName("dagger.assisted", "AssistedFactory")
+
+    val factoryBuilder =
+      TypeSpec.interfaceBuilder("Factory").apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedFactory)
+        }
+      }
+
+    val constructorBuilder =
+      FunSpec.constructorBuilder().apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedInject)
+        }
+        addParameter(
+          ParameterSpec.builder("screen", ClassName("", screenClassName))
```

Comment:
This will end up in a ClassName with no package. Can we pass in the original `Screen` classname directly?